src/query-compiler/default-query-compiler.ts

```diff
@@ -572,6 +576,83 @@ export class DefaultQueryCompiler
     }
   }
 
+  protected override visitCreateTrigger(node: CreateTriggerNode): void {
+    if (!node.time) throw new Error('Trigger time is required.')
+    if (!node.events) throw new Error('Trigger event is required.')
+    if (!node.table) throw new Error('Trigger table is required.')
```

Comment:
We generally don't do stuff like that.

- user brings his own SQL knowledge. let database error, user understand its errors.
- based on the WYSIWYG design principle, output query as-is.
- sometimes you can, with a more complex API design, enforce these props' existence before execution.

Reply:
> sometimes you can, with a more complex API implementation, enforce these props' existence before execution - making the API more foolproof.

That was my first thought as well. However, I wanted to make the API "nicer" with only one parameter in `createTrigger` instead of 3 as the code is more readable that way.